AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/RemoteSyncEngine.swift

```diff
@@ -365,6 +367,15 @@ class RemoteSyncEngine: RemoteSyncEngineBehavior {
         Amplify.Hub.dispatch(to: .dataStore, payload: networkStatusEventPayload)
     }
 
+    func cancelEmitter() {
+        if syncEventEmitter != nil {
+            syncEventEmitter = nil
+        }
+        if readyEventEmitter != nil {
+            readyEventEmitter = nil
+        }
```

Comment:
Nit, it's innocuous to assign these emitters to `nil` even if they're already `nil`, and avoids the (very unlikely) possibility of a race condition on the `if` check.
```suggestion
        syncEventEmitter = nil
        readyEventEmitter = nil
```

Reply:
Updated